yoke/assets/joypad/base.js

```diff
@@ -867,38 +875,27 @@ function loadPad(filename) {
         document.getElementById('joypad').style.display = 'grid';
         if (window.CSS && CSS.supports('display', 'grid')) {
             warningDiv.style.display = 'none';
-            if (minForce > maxForce) { // no touch force detection capability
+            if (PRESSURE_DETECTION_ENABLED == false || joypad == 0 || joypad == 1 || joypad == null) { // no touch force detection capability
                 joypad = new Joypad();
             } else { // possible force detection capability
-                var calibrationDiv = document.getElementById('calibration');
-                forceBar.style.opacity = '0.2';
-                calibrationDiv.style.display = 'inline';
-                calibrationDiv.addEventListener('touchmove', recordPressure);
-                calibrationDiv.addEventListener('touchend', function() { forceBar.style.opacity = '0.2'; });
-                document.getElementById('calibrationOk').addEventListener('click', function() {
-                    calibrationDiv.style.display = 'none';
-                    Pedal.prototype.onTouchMove = Pedal.prototype.onTouchMoveForce;
-                    AnalogButton.prototype.processTouches = AnalogButton.prototype.processTouchesForce;
-                    joypad = new Joypad();
-                });
-                document.getElementById('calibrationNo').addEventListener('click', function() {
-                    calibrationDiv.style.display = 'none';
-                    minForce = 1; maxForce = 0; joypad = new Joypad();
-                });
+                Pedal.prototype.onTouchMove = Pedal.prototype.onTouchMoveForce;
+                AnalogButton.prototype.processTouches = AnalogButton.prototype.processTouchesForce;
+                Joystick.prototype.checkThumbButton = Joystick.prototype.checkThumbButtonForce;
+                joypad = new Joypad();
             }
         }
     };
 
     head.appendChild(link);
 }
 
-var forceBar = document.getElementById('force');
 var warningDiv = document.getElementById('warning');
 document.getElementById('menu').childNodes.forEach(function(child) {
     var id = child.id;
     if (id) {
         child.addEventListener('click', function() { loadPad(id + '.css'); });
-        child.addEventListener('touchstart', recordPressure);
-        child.addEventListener('touchmove', recordPressure);
+        // If the pressure is between 0 (not normally achievable) or 1 (maximum),
+        // the touchscreen can almost certainly detect finger pressure.
+        child.addEventListener('touchstart', function(ev) { joypad = ev.targetTouches[0].force; });
```

Comment:
I have to admin I got a bit confused initially here because `joypad` variable is reused.

Do I assume correctly that the expected control flow is this:
- after the user selects a joypad in the menu, `touchstart` event is fired, which saves the force
- in the same tick `click` event is fired, which loads the joypad
- Joypad code sees that `joypad` variable is e.g. `0.42`, so it knows that force detection is available
- Joypad code sets `joypad` to an instance of Joypad class

Now, from what I understand, `touchstart` event always (?) comes before `click`, because click is `touchstart + touchend`.

User cannot go back to menu, so joypad being overriden again is also not a problem.

But I'm not sure if this code wouldn't be easier to understand if the pressure was just stored in a separate variable. Or maybe the lines 878-884 could be moved here, into `menu` logic, as the `joypad == 1` check looks weird there.

Reply:
Is this alright? 537187abcb68a2874b3fa65512a77634b964a2a4

Reply:
Oh, I forgot to answer your question. Yes, you're completely right about the control flow.

I added some comments along with your suggestion to make it clearer.

Reply:
Looks good to me. Thanks for the changes.